src/Turtle/Options.hs

```diff
@@ -0,0 +1,100 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
+-- TODO: documentation
+
+module Turtle.Options
+    ( Parser
+    , ParameterRead
+    , parameterRead
+    , ParameterName(..)
+    , LongName(..)
+    , HelpMessage(..)
+    , options
+    , switch
+    , parameter
+    , pAuto
+    , pText
+    , pInteger
+    , pDouble
+    ) where
+
+import Data.Monoid
+import Data.String (IsString)
+import Text.Read (readMaybe)
+import Data.Text (Text)
+import qualified Data.Text as Text
+import Data.Optional
+import Control.Applicative
+import Control.Monad.IO.Class
+import Control.Monad.Trans.Reader
+import Options.Applicative (Parser)
+import qualified Options.Applicative as Opts
+import qualified Options.Applicative.Types as Opts
+
+options :: MonadIO io => Text -> Parser a -> io a
+options header parser = liftIO
+    $ Opts.execParser
+    $ Opts.info (Opts.helper <*> parser) (Opts.header (Text.unpack header))
+
+newtype ParameterName = ParameterName { getParameterName :: Text }
+    deriving (IsString)
+
+newtype LongName = LongName { getLongName :: Text }
+    deriving (IsString)
+
+newtype ShortName = ShortName { getShortName :: Char }
+
+newtype HelpMessage = HelpMessage { getHelpMessage :: Text }
+    deriving (IsString)
+
+optionalOr :: Monoid m  => (a -> m) -> Optional a -> m
```

Comment:
`optionalOr` is the exact same thing as `Data.Foldable.foldMap` since `Optional` implements `Foldable`